lib/eslint.js

```diff
@@ -757,6 +757,8 @@ module.exports = (function() {
      */
     api.verify = function(textOrSourceCode, config, filenameOrOptions, saveState) {
         const text = (typeof textOrSourceCode === "string") ? textOrSourceCode : null;
+        const selectors = new Set();
```

Comment:
I haven't seen any benchmarks for `Set` insertion, but one possible additional perf improvement might be to not add every selector to the set, but instead grab all selectors from EventEmitter. EventEmitter has hidden property that contains a list of all of the subscribed events, so it already dedups it.